feat(templates): use uv for testing and packing kubernetes and machine charms (#2312)
canonical/charmcraft: docs/reference/files/uv-lock-file.rst

.. _uv-lock-file:

``uv.lock`` file
================

> The ``uv.lock`` file in your charm's root directory is a lockfile that specifies the

Reviewer: ```suggestion
The ``uv.lock`` file in your charm's root directory specifies the
```

Circular definition.

exact versions of your charm's dependencies and your development dependencies.

>     See more: `uv |
>     The lockfile
>     <https://docs.astral.sh/uv/concepts/projects/layout/#the-lockfile>`_

Reviewer: ```suggestion
.. seealso::

    `uv | The lockfile
    <https://docs.astral.sh/uv/concepts/projects/layout/#the-lockfile>`_
```

This file is required if your charm uses the :ref:`craft_parts_uv_plugin`. It's also
required if you use tox with tox-uv. For more information, see :ref:`tox-ini-file`.

Charmcraft doesn't create this file when you run the :ref:`ref_commands_init` command.

> For ``init`` with the ``kubernetes`` or ``machine`` profile, you should create this

Reviewer: ```suggestion
After initializing a charm with the Kubernetes or machine profile, create this
```

file by running ``uv lock``. Alternatively, you can create/update this file by using
``uv add`` and ``uv remove`` to manage dependencies in the :ref:`pyproject-toml-file`.

You shouldn't manually edit this file.

> For ``init`` with a 12-factor profile (``django-framework`` and so on), this file isn't

Reviewer: ```suggestion
For the 12-factor app profiles, this file isn't
```
